MyListLibrary/MyList.cs

```diff
@@ -0,0 +1,343 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace MyListLibrary
+{
+    public class MyList<T> : IEnumerable<T>, IMyList<T> where T : IComparable<T>
+    {
+        private const int DEFAULT_SIZE = 4;
+        private const double INCREAMENTOR = 1.33;
+
+        private T[] array;
+        public int Count { get; set; }
+        public int Capacity { get => array.Length; }
+
+        public MyList()
+        {
+            this.array = new T[DEFAULT_SIZE];
+        }
+
+        public MyList(T value)
+        {
+            this.array = new T[DEFAULT_SIZE];
+            array[0] = value;
+            ++Count;
+        }
+
+        public MyList(T[] array)
+        {
+            this.array = array;
```

Comment:
Here you're coping links, not array. If you will change this array outside the class, it will be changed here to. You need to copy it

Reply:
good point